wpilibNewCommands/src/main/java/edu/wpi/first/wpilibj2/command/CommandScheduler.java

```diff
@@ -274,11 +264,19 @@ public void run() {
     loopCache.poll();
     m_watchdog.addEpoch("buttons.run()");
 
-    m_inRunLoop = true;
     boolean isDisabled = RobotState.isDisabled();
     // Run scheduled commands, remove finished commands.
-    for (Iterator<Command> iterator = m_scheduledCommands.iterator(); iterator.hasNext(); ) {
-      Command command = iterator.next();
+    m_scheduledCommandsCopy = m_scheduledCommands.toArray(m_scheduledCommandsCopy);
+    for (Command command : m_scheduledCommandsCopy) {
+      if (command == null) {
+        // at least 1 Command was removed before `run` was called (diff between copy and original).
```

Comment:
```suggestion
```
This isn't the case- null should only be present in the array after the last element of the set

Reply:
Oops, it was stuck there from the previous implementation, thanks for pointing it out fixing it now.